vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/jsonitem/ArrayPointable.java

```diff
@@ -40,36 +41,13 @@ public IPointable createPointable() {
         }
     };
 
-    public int getEntryCount() {
-        return getEntryCount(bytes, start);
-    }
-
-    private static int getEntryCount(byte[] bytes, int start) {
-        return IntegerPointable.getInteger(bytes, start);
-    }
-
-    public void getEntry(int idx, IPointable pointer) {
-        int dataStart = getDataStart(bytes, start);
-        pointer.set(bytes, dataStart + getRelativeEntryStartOffset(idx), getEntryLength(idx));
-    }
-
-    private static int getEntryOffsetValue(byte[] bytes, int start, int idx) {
-        return IntegerPointable.getInteger(bytes, getOffsetsStart(start) + idx * SLOT_SIZE);
-    }
+    private TaggedValuePointable tvp = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
 
-    private int getRelativeEntryStartOffset(int idx) {
-        return idx == 0 ? 0 : getEntryOffsetValue(bytes, start, idx - 1);
-    }
-
-    private int getEntryLength(int idx) {
-        return getEntryOffsetValue(bytes, start, idx) - getRelativeEntryStartOffset(idx);
-    }
-
-    private static int getOffsetsStart(int start) {
-        return start + ENTRY_COUNT_SIZE;
-    }
-
-    private static int getDataStart(byte[] bytes, int start) {
-        return getOffsetsStart(start) + getEntryCount(bytes, start) * SLOT_SIZE;
+    public void appendSequence(SequenceBuilder sb) throws IOException {
```

Comment:
Should the title be different? You are not appending a sequence, but all the individual items in the sequence.